### ISSUE TITLE
Add oc set-deployment strategy command.

### ISSUE THREAD

**Issue by GrahamDumpleton** [openshift/origin#17462]
The only way I know of for changing deployment strategy from command line is using +oc edit+ or +oc patch+. There really should be an +oc set deployment-strategy+ option to allow the strategy to be changed, as well as set other parameters of +dc.spec.strategy+.

It is a common enough problem when only having ``RWO`` persistent volumes, of needing to swap the deployment strategy to ``Recreate`` to avoid pods hanging due to persistent volume not being able to be mounted against multiple nodes.

Being able to run:

```
oc set deployment-strategy dc/blog --strategy Recreate
```

is more understandable than having to run:

```
oc patch dc/blog --patch '{"spec":{"strategy":{"type":"Recreate"}}}'
```

**Comment by mfojtik** [comment]
Agree, I was thinking about this command for a long time. @kargakis @tnozicka agree this will be useful?

**Comment by kargakis** [comment]
Yes, we have been wanting this for some time now
